move/axelar_gateway/sources/gateway.move

```diff
@@ -228,24 +242,41 @@ entry fun rotate_signers(
 // Public Functions
 // ----------------
 
-/// Call a contract on the destination chain by sending an event from an
-/// authorized Channel. Currently we require Channel to be mutable to prevent
-/// frozen object scenario or when someone exposes the Channel to the outer
-/// world. However, this restriction may be lifted in the future, and having
-/// an immutable reference should be enough.
-public fun call_contract(
+/// Prepare a MessageTicket to call a contract on the destination chain.
+public fun prepare_message(
     channel: &Channel,
     destination_chain: String,
     destination_address: String,
     payload: vector<u8>,
+): MessageTicket {
+    MessageTicket{
+        source_id: channel.to_address(),
+        destination_chain,
+        destination_address,
+        payload,
+        version: VERSION,
+    }
+}
+
+/// Submit the MessageTicket which causes a contract call by sending an event from an
+/// authorized Channel.
+entry fun submit_message(
+    message: MessageTicket,
 ) {
+    let MessageTicket {
+        source_id,
+        destination_chain,
+        destination_address,
+        payload,
+        version,
+    } = message;
     sui::event::emit(ContractCall {
```

Comment:
Since this still requires Amplifier and relayers to update to the latest package on upgrade, can you see if they can be identified by the Gateway object instead? With the version control requirement, we could make it so that `submit_message` can only be called from the latest package and take in `&AxelarGateway`. Can this be tied to the event such that we don't need to check the package id? The event filter has an Object [query](https://docs.sui.io/guides/developer/sui-101/using-events#filtering-event-queries), can you investigate if that's useful?

Reply:
The event should be the same for all packages (assuming we don't need to emit additional info), so the relayers can easily query all of the events.

Reply:
Can you share example testnet txs with call contract events before and after an upgrade?

Reply:
Great, so relayers or amplifier can check against the same package id then